src/NetEscapades.EnumGenerators/SourceGenerationHelper.cs

```diff
@@ -962,4 +963,63 @@ public static string[] GetNames()
             .ToString();
         return (content, filename);
     }
+
+    private static IReadOnlyCollection<(string Key, EnumValueOption Value)> GetNamesOrderedByValue(EnumToGenerate enumToGenerate)
+    {
+        var valueComparer = Comparer<object>.Default;
```

Comment:
As mentioned above, using `object` here causes a bunch of boxing. I think we can improve this by just using `long` values